src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -530,7 +531,7 @@ public List<DarCollection> addDatasetsToCollections(List<DarCollection> collecti
       if (!filterDatasetIds.isEmpty()) {
         datasetIds.retainAll(filterDatasetIds);
       }
-      Set<Dataset> datasets = datasetDAO.findDatasetWithDataUseByIdList(datasetIds);
+      Set<Dataset> datasets = new HashSet<>(datasetDAO.findDatasetsByIdList(datasetIds));
       Map<Integer, Dataset> datasetMap = datasets.stream()
           .collect(Collectors.toMap(Dataset::getDatasetId, Function.identity()));
```

Comment:
The only use of `datasets` is to call `stream()`. If you want to remove duplicates from a `Stream` you can use `distinct()`.
```
      Set<Dataset> datasets = datasetDAO.findDatasetWithDataUseByIdList(datasetIds);
      Map<Integer, Dataset> datasetMap = datasets.stream()
          .distinct()
          .collect(Collectors.toMap(Dataset::getDatasetId, Function.identity()));
```

Reply:
Ah - good point, I was trying to keep the diff as small as possible here, but we really didn't need the `Set`. I can clean this up some more.